crates/sui-config/tests/snapshot_tests.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// To review and fix test failures in this file,
+// 0. Install cargo-insta
+// 1. Run `cargo insta test --review` under `./sui-config`.
+// 2. Review, accept or reject changes.
+
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use insta::assert_yaml_snapshot;
+use multiaddr::Multiaddr;
+use narwhal_crypto::traits::KeyPair;
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+use sui_config::{genesis::Builder, genesis_config::GenesisConfig};
+use sui_config::{NetworkConfig, ValidatorInfo};
+use sui_types::base_types::{ObjectID, SuiAddress};
+use sui_types::crypto::{get_key_pair_from_rng, PublicKeyBytes};
+
+#[test]
+fn genesis_config_snapshot_matches() {
+    // Test creating fake SuiAddress from PublicKeyBytes.
+    let keypair = get_key_pair_from_rng(&mut StdRng::from_seed([0; 32])).1;
+    let public_key = PublicKeyBytes::from(keypair.public());
+    let fake_addr = SuiAddress::from(&public_key);
+
+    let fake_obj_id = "0x0fa3e0b50000496a53bafcea629fcbcfff2a9866";
```

Comment:
You can actually get an `ObjectID` from an address, and in fact I think the best way to get it is to use `ObjectID::random()`.

Reply:
Changed to getting `ObjectID` from the fake address. The object ID needs to be static because it is part of the snapshot.

Reply:
Of course! makes sense.